lib/netstd/Thrift/Transport/Client/THttpTransport.cs

```diff
@@ -33,78 +33,73 @@ public class THttpTransport : TTransport
         private readonly X509Certificate[] _certificates;
         private readonly Uri _uri;
 
-        // Timeouts in milliseconds
-        private int _connectTimeout = 30000;
+        private int _connectTimeout = 30000; // Timeouts in milliseconds
         private HttpClient _httpClient;
         private Stream _inputStream;
-
+        private MemoryStream _outputStream;
         private bool _isDisposed;
-        private MemoryStream _outputStream = new MemoryStream();
 
-        public THttpTransport(Uri u, IDictionary<string, string> customHeaders = null, string userAgent = null)
-            : this(u, Enumerable.Empty<X509Certificate>(), customHeaders, userAgent)
+        public THttpTransport(Uri uri, IDictionary<string, string> customRequestHeaders = null, string userAgent = null)
+            : this(uri, Enumerable.Empty<X509Certificate>(), customRequestHeaders, userAgent)
         {
         }
 
-        public THttpTransport(Uri u, IEnumerable<X509Certificate> certificates,
-            IDictionary<string, string> customHeaders, string userAgent = null)
+        public THttpTransport(Uri uri, IEnumerable<X509Certificate> certificates,
+            IDictionary<string, string> customRequestHeaders, string userAgent = null)
         {
-            _uri = u;
+            _uri = uri;
             _certificates = (certificates ?? Enumerable.Empty<X509Certificate>()).ToArray();
-            CustomHeaders = customHeaders;
 
             if (!string.IsNullOrEmpty(userAgent))
                 UserAgent = userAgent;
 
             // due to current bug with performance of Dispose in netcore https://github.com/dotnet/corefx/issues/8809
             // this can be switched to default way (create client->use->dispose per flush) later
-            _httpClient = CreateClient();
+            _httpClient = CreateClient(customRequestHeaders);
         }
 
         // According to RFC 2616 section 3.8, the "User-Agent" header may not carry a version number
         public readonly string UserAgent = "Thrift netstd THttpClient";
 
-        public IDictionary<string, string> CustomHeaders { get; }
+        public override bool IsOpen => _outputStream != null;
 
-        public int ConnectTimeout
-        {
-            set { _connectTimeout = value; }
-        }
+        public HttpRequestHeaders RequestHeaders => _httpClient.DefaultRequestHeaders;
 
-        public override bool IsOpen => true;
+        public MediaTypeHeaderValue ContentType { get; set; }
 
-        public override async Task OpenAsync(CancellationToken cancellationToken)
+        public override Task OpenAsync(CancellationToken cancellationToken)
         {
-            if (cancellationToken.IsCancellationRequested)
+            if (_isDisposed)
```

Comment:
Could you comment on that change? Why do we need that and how is it related to setting custom HTTP headers?